### ISSUE TITLE
Kopia docker - /bin/fusermount: no such file or directory

### ISSUE THREAD

**Issue by marcosvrs** [kopia/kopia#2191]
Can't mount an existing snapshot in docker. fusermount is missing on docker container.

There are no logs, but the response of the POST request:
XHRPOSThttps://kopiasecreturl/api/v1/mounts
[HTTP/2 500 Internal Server Error 133ms]
```json
{
  "code": "INTERNAL",
  "error": "internal server error: unable to mount: mounting error: exec: \"/bin/fusermount\": stat /bin/fusermount: no such file or directory"
}
```

**Comment by Yanik39** [comment]
Any solution?

**Comment by jkowalski** [comment]
So what were you expecting to see? Even if you mount a file inside a docker container it won't be accessible outside, right? What's your use case?

**Comment by Yanik39** [comment]
We can bind something like /mnt/kopia to docker and mount snapshots and we can reach them outside the docker.
What does mean use case? its same with use case of kopia.

**Comment by jkowalski** [comment]
This explains the issue more:

https://github.com/kopia/kopia/issues/2238

PRs to add a flavor of kopia Docker image based on some real OS are welcome.

**Comment by marcosvrs** [comment]
Why don’t you include fusermount binary the same as the rclone?
Or even easier, let we bind the current fusermount from our host OS and you don’t need to provide anything else.

Our goal here is to recover our backup from our external repos via the container.
I assume this is a important feature for a backup manager tool like Kopia and should not be “worked around”… I just want to be able to recover my files :(

**Comment by jkowalski** [comment]
Kopia is primarily designed to be used without containers - it's a single statically linked binary file you can just run without worrying about such dependencies. 

Container image is provided for folks who must run containers in their environment, but if you have a choice, most of the time there are no benefits to running as containers (given the complexity of figuring out mounts, permissions, config files, parameters)

**Comment by marcosvrs** [comment]
I see, you basically say that Kopia container shouldn’t be used at all…
Message understood, but not agreed.

Nevertheless, I already created my own Docker image to work with Kopia.

Kind Regards
Marcos Vinícius Rubido
________________________________

**Comment by jkowalski** [comment]
Not saying it should't be used, it's easier to use it without it.

**Comment by marcosvrs** [comment]
We were supposed to be discussing the possible solution how to solve the issue, and not explaining to us why not use the Docker image.
If you are not able to support the Docker image, I suggest you to remove it from the Docker hub. Otherwise you will continue to receive the bug reports from it.

Kind Regards
Marcos Vinícius Rubido
________________________________

**Comment by Yanik39** [comment]
> We were supposed to be discussing the possible solution how to solve the issue, and not explaining to us why not use the Docker image. If you are not able to support the Docker image, I suggest you to remove it from the Docker hub. Otherwise you will continue to receive the bug reports from it. Kind Regards Marcos Vinícius Rubido

i suggest not to remove dockerfile.  just few words at readme would be enough.

**Comment by jkowalski** [comment]
I think we should publish another docker image based on Ubuntu or Alpine as an option to users who need it and document it. 

@marcosvrs do you want to contribute your Dockerfile and help maintain it for other Kopia users? I can help integrate it into CI/CD.

**Comment by Yanik39** [comment]
Meanwhile i build it on ubuntu:jammy with proper docker-compose and docker KopiaUI mounts are visible at host if needed.
https://github.com/Yanik39/Kopia-Ubuntu

**Comment by jkowalski** [comment]
@Yanik39 this is great, would you be interested in maintaining your images long-term for other users? If so, I'd be happy to link to your repo from kopia.io? Or would you perhaps want to contribute this to Kopia so we make Ubuntu-based in addition to distroless ones on each release?

**Comment by Yanik39** [comment]
Thanks @jkowalski.
I can maintain but i think would be better if kopia has ubuntu based distro directly.
You may get my dockerfile and necessary files from my repo.

**Comment by jkowalski** [comment]
@Yanik39 Do you think you can make a PR that adds your Dockerfile to https://github.com/kopia/kopia/blob/master/tools/docker-publish.sh ?

Would it make sense to have 2 separate images (kopia/kopia:XXX and kopia-kopia:ubuntu-XXX) or should we only have ubuntu-based one?

**Comment by Yanik39** [comment]
Sure i can make a PR.

But better to add second Dockerfile as ubuntu.Dockerfile with my Dockerfile.
My build is based on kopia/kopia:latest. So better to keep original Dockerfile/kopia container.
Also it gives you more space to edit/change original Dockerfile.

**Comment by jkowalski** [comment]
Ok, I'm markins this issue as Fixed with #2252
